sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+
+        Initialization involves fetching bucket if exists, or creating it if
+        it does not. Also, it checks for the existance of the storage account
+        and resource group if provided by the user. If not provided, those are
+        created with a default naming convention.
+
+        Raises:
+            StorageBucketCreateError: If container creation fails or storage
+                account attempted to be created already exists.
+            StorageBucketGetError: If fetching existing bucket fails.
+            StorageInitError: If general initialization fails.
+        """
+        self.storage_client = data_utils.create_az_client('storage')
+        self.resource_client = data_utils.create_az_client('resource')
+        # Using externally created storage
+        if isinstance(self.source, str
+                      ) and data_utils.is_az_container_endpoint(self.source):
+            bucket_name, _, storage_account_name = data_utils.split_az_path(
+                self.source)
+            assert self.name == bucket_name
+            self.storage_account_name = storage_account_name
+            self.resource_group_name = data_utils.get_az_resource_group(
+                self.storage_account_name)
+        # Creates new resource group and storage account or use the 
+        # storage_account provided by the user through config.yaml
+        else:
+            config_storage_account = skypilot_config.get_nested(
+                ('azure', 'storage_account'), None)
+            if config_storage_account is not None:
+                # using user provided storage account from config.yaml
+                self.storage_account_name = config_storage_account
+                self.resource_group_name = data_utils.get_az_resource_group(
+                    self.storage_account_name)
+                # when the provided storage account does not exist under user's
+                # subscription id.
+                if self.resource_group_name is None:
+                    with ux_utils.print_exception_no_traceback():
+                        raise ValueError(
+                            'The storage account '
+                            f'{self.storage_account_name!r} specified in '
+                            'config.yaml does not exist under the user\'s '
+                            'subscription ID. Provide a storage account '
+                            'through config.yaml only when creating a '
+                            'container under an already existing storage '
+                            'account within your subscription ID.'
+                        )
+            else:
+                # If storage account name is not provided from config, then 
+                # use default resource group and storage account names.
+                self.storage_account_name = f'sky{self.region}{common_utils.get_user_hash()}'
+                self.resource_group_name = f'sky{common_utils.get_user_hash()}'
```

Comment:
feeling like we should have a constant for those two default names.

Reply:
Agreed! Added at [aea8316](https://github.com/skypilot-org/skypilot/pull/3032/commits/aea8316d7a5ae0652bf565e823f243df545c6231)